userspace/libsinsp/filterchecks.cpp

```diff
@@ -1406,76 +1406,44 @@ bool sinsp_filter_check_fd::compare_ip(sinsp_evt *evt)
 
 bool sinsp_filter_check_fd::compare_net(sinsp_evt *evt)
 {
-	if(!extract_fd(evt))
+	if(!extract_fd(evt) || m_fdinfo == nullptr)
 	{
 		return false;
 	}
 
-	if(m_fdinfo != NULL)
+	bool sip_cmp = false;
+	bool dip_cmp = false;
+
+	switch (m_fdinfo->m_type)
 	{
-		scap_fd_type evt_type = m_fdinfo->m_type;
+	case SCAP_FD_IPV4_SERVSOCK:
+		return flt_compare_ipv4net(m_cmpop, m_fdinfo->m_sockinfo.m_ipv4serverinfo.m_ip, (ipv4net*)filter_value_p());
 
-		if(evt_type == SCAP_FD_IPV4_SOCK)
-		{
-			if(m_cmpop == CO_EQ || m_cmpop == CO_IN)
-			{
-				if(flt_compare_ipv4net(m_cmpop, m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip, (ipv4net*)filter_value_p()) ||
-				   flt_compare_ipv4net(m_cmpop, m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip, (ipv4net*)filter_value_p()))
-				{
-					return true;
-				}
-			}
-			else if(m_cmpop == CO_NE)
-			{
-				if(flt_compare_ipv4net(m_cmpop, m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip, (ipv4net*)filter_value_p()) &&
-				   flt_compare_ipv4net(m_cmpop, m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip, (ipv4net*)filter_value_p()))
-				{
-					return true;
-				}
-			}
-			else
-			{
-				throw sinsp_exception("filter error: IP filter only supports '=' and '!=' operators");
```

Comment:
no change in behavior compared to the old one: here is the exception same as in  flt_compare_ipv4net - just duplicated code here.